### PR TITLE
fix(Formula AST): Be more permissive on column names 

### PR DESCRIPTION
**Pull request by lukapeschke**
Allow column names to contain operators and extra whitespaces.

An implementation change is required for that: We now store the start and end position of bracket
delimiters and consider everything in-between to be the column name (meaning we don't strip whitespace
at the start and end anymore, nor do we check the validity of the tokens in the column name).

TCTC-3866

### CODE REVIEW

File: server/src/weaverbird/pipeline/formula_ast/eval.py [hunk 1]
```diff
@@ -63,21 +63,19 @@ def next_token():
                 return None
 
         def parse_col_name(prev_token: tokenize.TokenInfo) -> str:
-            chunks = []
+            # NOTE: position is expressed as a (posLine, posColumn) tupple. We assume that the
```
Comment (lukapeschke) [inline server/src/weaverbird/pipeline/formula_ast/eval.py:66]:
```suggestion
            # NOTE: position is expressed as a (posLine, posColumn) tuple. We assume that the
```

File: server/src/weaverbird/pipeline/formula_ast/eval.py [hunk 2]
```diff
@@ -63,21 +63,19 @@ def next_token():
                 return None
 
         def parse_col_name(prev_token: tokenize.TokenInfo) -> str:
-            chunks = []
+            # NOTE: position is expressed as a (posLine, posColumn) tuple. We assume that the
+            # formula is written on a single line
+            start = prev_token.end[1]
```
Comment (Sanix-Darker) [inline server/src/weaverbird/pipeline/formula_ast/eval.py:68]: 👍🏽

**Review by Sanix-Darker — APPROVED** [review]
LGTM !

File: server/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+## [0.24.1] - 2022-09-13
+
+- Formula AST: Change column name parser implementation to be more permissive with column names
+  (allow columns to include dashes)
```
Comment (PrettyWood) [inline server/CHANGELOG.md:8]: Or quotes, white spaces...

File: server/src/weaverbird/backends/pandas_executor/steps/utils/formula.py [hunk 4]
```diff
@@ -29,9 +29,12 @@ def _eval_expression(df: DataFrame, expr: Expression) -> Series:
         return df[expr.name]
     elif isinstance(expr, str):
         # we want unquoted strings
-        expr = literal_eval(expr)
+        expr = unquote_string(expr, escape_quotes=False)
```
Comment (PrettyWood) [inline server/src/weaverbird/backends/pandas_executor/steps/utils/formula.py:32]: Deserves a comment

**Review by PrettyWood — CHANGES_REQUESTED** [review]

**Review by PrettyWood — APPROVED** [review]

**Pull request merged**